From 3257b73483e30abf170653a54818c5ee32693f73 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 02 May 2019 04:54:01 +0300
Subject: [PATCH] [SW-0] pll: do per platform handling

In ONIX, and PLDM it is only an empty stub which returns that no PLLs
available.

Change-Id: I7fae4f9be39b3fbbdfeb4b8e5957c4c0b3248475
---

diff --git a/Makefile b/Makefile
index d500a49..7d84dbc 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,8 @@
 LDLIBS := -lgcc
 
 PREBOOT_SRCS_C := $(shell ls common/*.c)
-PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/*.c)
+PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/common.c)
+PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/$(HL_TARGET).c)
 PREBOOT_SRCS_S := $(shell ls common/*.S)
 
 PREBOOT_OBJECTS := $(PREBOOT_SRCS_C:.c=.o)
diff --git a/gaudi/asic.c b/gaudi/asic.c
new file mode 100644
index 0000000..9cb7f88
--- /dev/null
+++ b/gaudi/asic.c
@@ -0,0 +1,117 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include "common.h"
+
+#include <psoc_global_conf_regs.h>
+
+#include <preboot.h>
+#include <pll.h>
+
+static struct hl_pll_cfg gaudi_pll_cfg[] = {
+		{
+			.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
+	//		.nr = 0, .nf = 47, .od = 1, .nb = 23,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 47, -1 }
+		}, {
+			.name = "DMA_W", .base = mmDMA_W_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "IF_W", .base = mmIF_W_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "IF_E", .base = mmIF_E_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
+	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, -1, -1, -1 }
+		}, {
+			.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
+	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, -1, -1, -1 }
+		}, {
+			.name = "MME", .base = mmPSOC_MME_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
+	//		.nr = 0, .nf = 37, .od = 1, .nb = 18,
+			.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
+			.div_factor = { 0, 0, 4, 9 }
+		}, {
+			.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "HBM", .base = mmPSOC_HBM_PLL_BASE,
+			.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "NIC0", .base = mmNIC0_PLL_BASE,
+			.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
+			.div_factor = { 0, 4, 8, -1 }
+		}, {
+			.name = "NIC1", .base = mmNIC1_PLL_BASE,
+			.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
+			.div_factor = { 0, 4, 8, -1 }
+		}, {
+			.name = "NIC2_HBM", .base = mmNIC2_HBM_PLL_BASE,
+			.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "NIC2_MME", .base = mmNIC2_MME_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		},
+};
+
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
+{
+	int pll_num = ARRAY_SIZE(gaudi_pll_cfg);
+	int boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
+
+	if (boot_strap & SKIP_PLL_INIT)
+		return 0;
+
+	*hl_pll_cfg = gaudi_pll_cfg;
+
+	return pll_num;
+}
+
diff --git a/gaudi/common.c b/gaudi/common.c
index 6fbbfa3..f7a61c9 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -60,109 +60,6 @@
 	return 1200000000ULL / USEC_PER_SEC;
 }
 
-static struct hl_pll_cfg gaudi_pll_cfg[] = {
-		{
-			.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
-	//		.nr = 0, .nf = 47, .od = 1, .nb = 23,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 47, -1 }
-		}, {
-			.name = "DMA_W", .base = mmDMA_W_PLL_BASE,
-	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, -1, -1 }
-		}, {
-			.name = "IF_W", .base = mmIF_W_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "IF_E", .base = mmIF_E_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
-	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, -1, -1 }
-		}, {
-			.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
-	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
-	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, -1, -1, -1 }
-		}, {
-			.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
-	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
-	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, -1, -1, -1 }
-		}, {
-			.name = "MME", .base = mmPSOC_MME_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
-	//		.nr = 0, .nf = 37, .od = 1, .nb = 18,
-			.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
-			.div_factor = { 0, 0, 4, 9 }
-		}, {
-			.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "HBM", .base = mmPSOC_HBM_PLL_BASE,
-			.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
-			.div_factor = { 0, 1, -1, -1 }
-		}, {
-			.name = "NIC0", .base = mmNIC0_PLL_BASE,
-			.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
-			.div_factor = { 0, 4, 8, -1 }
-		}, {
-			.name = "NIC1", .base = mmNIC1_PLL_BASE,
-			.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
-			.div_factor = { 0, 4, 8, -1 }
-		}, {
-			.name = "NIC2_HBM", .base = mmNIC2_HBM_PLL_BASE,
-			.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
-			.div_factor = { 0, 1, -1, -1 }
-		}, {
-			.name = "NIC2_MME", .base = mmNIC2_MME_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		},
-};
-
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	int pll_num = ARRAY_SIZE(gaudi_pll_cfg);
-	int boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
-
-	if (boot_strap & SKIP_PLL_INIT)
-		return 0;
-
-	*hl_pll_cfg = gaudi_pll_cfg;
-
-	return pll_num;
-}
-
 void hl_uart_init(void)
 {
 	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
diff --git a/gaudi/onix.c b/gaudi/onix.c
new file mode 100644
index 0000000..85badcc
--- /dev/null
+++ b/gaudi/onix.c
@@ -0,0 +1,14 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include <preboot.h>
+#include <pll.h>
+
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
+{
+	return 0;
+}
diff --git a/gaudi/pldm.c b/gaudi/pldm.c
new file mode 100644
index 0000000..85badcc
--- /dev/null
+++ b/gaudi/pldm.c
@@ -0,0 +1,14 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include <preboot.h>
+#include <pll.h>
+
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
+{
+	return 0;
+}
diff --git a/goya/asic.c b/goya/asic.c
new file mode 100644
index 0000000..8221909
--- /dev/null
+++ b/goya/asic.c
@@ -0,0 +1,37 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include "common.h"
+
+#include <preboot.h>
+#include <pll.h>
+
+static struct hl_pll_cfg goya_pll_cfg[] = {
+	{
+		.name = "CPU", .base = mmCPU_PLL_BASE,
+		.nr = 2, .nf = 159, .od = 1, .nb = 79,
+		.div_factor = { 0, -1, 1, -1 }
+	}, {
+		.name = "IC", .base = mmIC_PLL_BASE,
+		.nr = 0, .nf = 63, .od = 15, .nb = 31,
+		.div_factor = { 0, 1, 1, 2 }
+	}, {
+		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
+		.nr = 0, .nf = 63, .od = 15, .nb = 31,
+		.div_factor = { 0, 1, 2, 8 }
+	},
+};
+
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
+{
+	int pll_num = ARRAY_SIZE(goya_pll_cfg);
+
+	*hl_pll_cfg = goya_pll_cfg;
+
+	return pll_num;
+}
+
diff --git a/goya/common.c b/goya/common.c
index d8a76a3..b293a3b 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -187,31 +187,6 @@
 	return 1330000000ULL / USEC_PER_SEC;
 }
 
-static struct hl_pll_cfg goya_pll_cfg[] = {
-	{
-		.name = "CPU", .base = mmCPU_PLL_BASE,
-		.nr = 2, .nf = 159, .od = 1, .nb = 79,
-		.div_factor = { 0, -1, 1, -1 }
-	}, {
-		.name = "IC", .base = mmIC_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 1, 2 }
-	}, {
-		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 2, 8 }
-	},
-};
-
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	int pll_num = ARRAY_SIZE(goya_pll_cfg);
-
-	*hl_pll_cfg = goya_pll_cfg;
-
-	return pll_num;
-}
-
 void hl_uart_init(void)
 {
 	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
diff --git a/goya/onix.c b/goya/onix.c
new file mode 100644
index 0000000..85badcc
--- /dev/null
+++ b/goya/onix.c
@@ -0,0 +1,14 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include <preboot.h>
+#include <pll.h>
+
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
+{
+	return 0;
+}
diff --git a/goya/pldm.c b/goya/pldm.c
new file mode 100644
index 0000000..85badcc
--- /dev/null
+++ b/goya/pldm.c
@@ -0,0 +1,14 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include <preboot.h>
+#include <pll.h>
+
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
+{
+	return 0;
+}
